vendor/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -124,19 +120,30 @@ func WithMaxInFlightLimit(
 	nonMutatingLimit int,
 	mutatingLimit int,
 	longRunningRequestCheck apirequest.LongRunningRequestCheck,
+	watermarkEnabled bool,
 ) http.Handler {
 	if nonMutatingLimit == 0 && mutatingLimit == 0 {
 		return handler
 	}
 	var nonMutatingChan chan bool
 	var mutatingChan chan bool
-	if nonMutatingLimit != 0 {
-		nonMutatingChan = make(chan bool, nonMutatingLimit)
-		watermark.readOnlyObserver.SetDenominator(float64(nonMutatingLimit))
-	}
-	if mutatingLimit != 0 {
-		mutatingChan = make(chan bool, mutatingLimit)
-		watermark.mutatingObserver.SetDenominator(float64(mutatingLimit))
+
+	if watermarkEnabled {
```

Comment:
It seems some methods expect the `inFlightWatermark` to be a globally instantiated variable, it might be easier to have the `requestWatermark` be instantiated how it previously was but adding a new attribute of `enabled bool`

```golang
type requestWatermark struct {
        enabled                             bool
	phase                                string
	readOnlyObserver, mutatingObserver   fcmetrics.RatioedChangeObserver
	lock                                 sync.Mutex
	readOnlyWatermark, mutatingWatermark int
}
```

and set it here as `inFlightWatermakr.enabled = watermarkEnabled` and then utilize that. 

What do you think?

Reply:
On the other hand, doing it that way it's implemented here avoids instantiating some histograms that won't end up being used.